src/fsharp/tast.fs

```diff
@@ -3020,15 +3031,14 @@ and
       mutable binding: NonNullSlot<Val>
       /// Indicates a reference to something bound in another CCU 
       nlr: NonLocalValOrMemberRef }
-    member x.IsLocalRef = match box x.nlr with null -> true | _ -> false
-    member x.IsResolved = match box x.binding with null -> false | _ -> true
+    member x.IsLocalRef = obj.ReferenceEquals(x.nlr, null)
```

Comment:
Please double check this change for performance or at least the generated IL

``match box x with null -> ... | ...`` should, AFAIK, generate the best code, and should also allow optimization to ``brfalse``  conditiona sitching in the IL.   There's no need to remove it from the compiler - if it's not generating good code we should fix our codegen.